### ISSUE TITLE
[OS X] Incorrectly detects a local session as telnet

### ISSUE THREAD

**Issue by daverigby** [nojhan/liquidprompt#407]
See https://github.com/nojhan/liquidprompt/issues/304#issuecomment-156447256 - as of commit 0d7d3fdcf1a892149c0ffbbc580248deefe0be70 the hostname is incorrectly detected as telnet.

Environment: OS X 10.11 (El Capitan).

**Comment by nedbat** [comment]
This is also happening to me on Yosemite.  This is the output of the commands you're using to decide:
```
$ LANG=C who am i
ned      ttys004  Nov 14 08:08
$ ps -o comm= -p $PPID 2> /dev/null
login
```

**Comment by dolmen** [comment]
Please try the `fix-407` branch: 119477c6da720e5cddb62940e469e1c602906ae7

**Comment by nedbat** [comment]
The fix-407 branch works properly for me.

**Comment by dolmen** [comment]
> The fix-407 branch works properly for me.

Great! I've merged the commit on `develop` with cf015ea399d200b698748c8eae7b0a2fc69f4693.
